utils/eth1.go

```diff
@@ -309,103 +310,117 @@ func FormatAddressLong(address string) template.HTML {
+func FormatAmountFormatted(amountInWei *big.Int, targetCurrency string, maxFractionalDigitsBeforeTrim int, maxIntegerDigitsBeforeTrim int, fullAmountTooltip bool, smallUnit bool, newLineForUnit bool) template.HTML {
+	return convertAndFormatWei(amountInWei, targetCurrency, maxFractionalDigitsBeforeTrim, maxIntegerDigitsBeforeTrim, fullAmountTooltip, smallUnit, newLineForUnit)
 }
-func FormatAmount(amount *big.Int, unit string, digits int) template.HTML {
-	return formatAmount(amount, unit, digits, 0, true, false, false)
+func FormatAmount(amountInWei *big.Int, targetCurrency string, maxFractionalDigitsBeforeTrim int) template.HTML {
+	return convertAndFormatWei(amountInWei, targetCurrency, maxFractionalDigitsBeforeTrim, 0, true, false, false)
 }
-func FormatBigAmount(amount *hexutil.Big, unit string, digits int) template.HTML {
-	return FormatAmount((*big.Int)(amount), unit, digits)
+func FormatBigAmount(amountInWei *hexutil.Big, targetCurrency string, maxFractionalDigitsBeforeTrim int) template.HTML {
+	return FormatAmount((*big.Int)(amountInWei), targetCurrency, maxFractionalDigitsBeforeTrim)
 }
-func FormatBytesAmount(amount []byte, unit string, digits int) template.HTML {
-	return FormatAmount(new(big.Int).SetBytes(amount), unit, digits)
+func FormatBytesAmount(amountInWei []byte, targetCurrency string, maxFractionalDigitsBeforeTrim int) template.HTML {
+	return FormatAmount(new(big.Int).SetBytes(amountInWei), targetCurrency, maxFractionalDigitsBeforeTrim)
 }
-func formatAmount(amount *big.Int, unit string, digits int, maxPreCommaDigitsBeforeTrim int, fullAmountTooltip bool, smallUnit bool, newLineForUnit bool) template.HTML {
-	// define display unit & digits used per unit max
-	displayUnit := " " + unit
-	var unitDigits int
-	if unit == "ETH" || unit == "Ether" || unit == "xDAI" || unit == "GNO" {
-		unitDigits = 18
-	} else if unit == "GWei" {
-		unitDigits = 9
-	} else {
-		displayUnit = " ?"
-		unitDigits = 0
+
+func convertAndFormatWei(amountInWei *big.Int, targetCurrency string, maxFractionalDigitsBeforeTrim int, maxIntegerDigitsBeforeTrim int, fullAmountTooltip bool, smallUnit bool, newLineForUnit bool) template.HTML {
+	var commaPositionFromEnd int
+	var targetAmount *big.Int
+
+	targetAmount = amountInWei
+	commaPositionFromEnd = price.ETHWeiCommaShift
+
+	if targetCurrency == "Ether" {
+		targetCurrency = "ETH"
+	}
+	if targetCurrency == "ETH" && amountInWei.Cmp(big.NewInt(1000*price.GWeiInWei)) <= 0 {
+		// The displayed unit will be GWei if the amount is small
+		targetCurrency = "GWei"
+	}
+
+	if targetCurrency == "GWei" {
+		commaPositionFromEnd = price.GWeiWeiCommaShift
+	} else if targetCurrency != "xDAI" && targetCurrency != "GNO" {
+		// The currency is a fiat. We convert the amount of Weis into it:
+		converted := new(big.Float)
+		converted = converted.Mul(new(big.Float).SetInt(amountInWei), big.NewFloat(price.GetPrice(Config.Frontend.ElCurrency, targetCurrency)))
```

Comment:
**nitpick:** A new line for `converted` doesn't really reduce the code in the next line.
I would rather create something for the price like
```suggestion
		price := big.NewFloat(price.GetPrice(Config.Frontend.ElCurrency, targetCurrency))
		converted := new(big.Float).Mul(new(big.Float).SetInt(amountInWei), price)
```

Reply:
OK, done.